File: Proiect/src/main/java/service/MasterService.java
package service;

import model.Match;

public class MasterService {
    private ServiceMatch matchService;
    private ServiceOrganiser organiserService;
    private ServiceTicket ticketService;

    public MasterService(ServiceMatch matchService, ServiceOrganiser organiserService, ServiceTicket ticketService) {
        this.matchService = matchService;
        this.organiserService = organiserService;
        this.ticketService = ticketService;
    }

    public void sellTicketForMatch(Integer mid,Integer quantity,String customerName){
        Match match = matchService.findMatchById(mid);
        int oldNoOfSeats = match.getNrOfSeats();
        if(quantity <= oldNoOfSeats) {
            Integer qt = oldNoOfSeats - quantity;
            matchService.updateMatchNoOfSeats(qt,mid);
            ticketService.saveTicket(mid,quantity,customerName);
        }
    }

    public ServiceMatch getMatchService() {
        return matchService;
    }

    public ServiceOrganiser getOrganiserService() {
        return organiserService;
    }

    public ServiceTicket getTicketService() {
        return ticketService;
    }
}

File: Csharp/MPP/Server/MasterService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using model;

namespace Server
{
    public class MasterService
    {
        public MasterService(ServiceMatch matchService, ServiceOrganiser organiserService, ServiceTicket ticketService)
        {
            MatchService = matchService;
            OrganiserService = organiserService;
            TicketService = ticketService;
        }

        public void sellTicketForMatch(int mid, int quantity, string customerName)
        {
            Match match = MatchService.findMatchById(mid);
            int oldNoOfSeats = match.NrOfSeats;
            if (quantity <= oldNoOfSeats)
            {
                int qt = oldNoOfSeats - quantity;
                MatchService.updateMatchNoOfSeats(qt, mid);
                TicketService.saveTicket(mid, quantity, customerName);
            }
        }

        public ServiceMatch MatchService
        {
            get;
            set;
        }

        public ServiceOrganiser OrganiserService
        {
            get;
            set;
        }

        public ServiceTicket TicketService
        {
            get;
            set;
        }
    }
}
